patch_discriminator.py

```diff
@@ -10,25 +10,24 @@ class PatchDiscriminator(nn.Module):
         image as a set of reference patches
 
         Takes a random sized patch (1/8 to 1/4 of the full image side-dimension up to 256) and outputs a scalar value
+
+        All patches should be scaled to the same size (patch_dim) (1/4 of image side-dimension) before being passed to
+        the discriminator -- Might need to be a power of 2, default: patch_dim=128
     """
     scale_up_to = 1024  # Patch size (dim*channels) to scale up to
     mc = 256 + 128  # Encoder output (maximum) number of channels to downscale to
 
     def __init__(self, patch_dim=128, channel_scale_factor=4.0):
         super().__init__()
 
-        # Since the patches have random sizes, they are upscaled to the same size before being passed to the
-        # co-occurrence discriminator
+        # Convolve 3 channels into a power of 2
         num_down_samples = int(math.ceil(math.log(patch_dim, 2)))
         scale_channel_amount = int(self.scale_up_to / (2 ** num_down_samples) * channel_scale_factor)
 
-        current_dim = patch_dim
-
         downscalers = []
         in_channels = scale_channel_amount
-        while current_dim > 4:
+        for sample in range(num_down_samples, 2, -1):
             out_channels = min(in_channels * 2, self.mc)
```

Comment:
Sorry for perhaps a silly question, but why is this multiplied by 2 here?

Reply:
It doubles the number of channels each time to reduce the layer dimensions by half
So a 256x256x2 input => 128x128x4 => 64x64x8 => 32x32x16 => 16x16x32 => ...